chromium_src/third_party/blink/renderer/core/html/canvas/canvas_async_blob_creator.cc

```diff
@@ -5,16 +5,13 @@
 
 #include "third_party/blink/public/platform/web_content_settings_client.h"
 #include "third_party/blink/renderer/core/execution_context/execution_context.h"
-#include "third_party/blink/renderer/core/frame/local_dom_window.h"
-#include "third_party/blink/renderer/core/frame/local_frame.h"
-#include "third_party/blink/renderer/core/workers/worker_global_scope.h"
 
-#define BRAVE_CANVAS_ASYNC_BLOB_CREATOR                                       \
-  if (WebContentSettingsClient* settings =                                    \
-          brave::GetContentSettingsClientFor(context)) {                      \
-    image_ = brave::BraveSessionCache::From(*context).PerturbPixels(settings, \
-                                                                    image_);  \
-    DCHECK(image_);                                                           \
+#define BRAVE_CANVAS_ASYNC_BLOB_CREATOR                                \
+  if (WebContentSettingsClient* settings =                             \
+          brave::GetContentSettingsClientFor(context_)) {              \
+    brave::BraveSessionCache::From(*context_).PerturbPixels(           \
+        settings, static_cast<const unsigned char*>(src_data_.addr()), \
```

Comment:
it has `writable_addr()`

Reply:
note that this object can have different `SkColorType`, so I'm not sure whether we can make any assumptions about pixels layout inside this

Reply:
cc @pilgrim-brave because the bytes per pixel was based on the existing code

Reply:
re `writable_addr` - I don't think that really matters here because it's just a const cast and we do a const cast internally and we don't always have a non-const reference we can pass in